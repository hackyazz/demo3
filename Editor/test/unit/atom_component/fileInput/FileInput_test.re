open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("FileInput", () => {
    let sandbox = getSandboxDefaultVal();

    beforeEach(() => sandbox := createSandbox());
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("test workflow", () => {
      test("set showInput button text", () =>
        ReactTestRenderer.create(<FileInput buttonText="showInput" />)
        |> ReactTestTool.createSnapshotAndMatch
      );
      test(
        "click the showInput button, show the textarea and submit-button", () => {
        let component =
          ReactTestRenderer.create(
            <FileInput isShowInput=true buttonText="showInput" />,
          );

        component |> ReactTestTool.createSnapshotAndMatch;
      });
      test("key in text", () => {
        let component =
          ReactTestRenderer.create(
            <FileInput
              inputValue="you can input value in textarea"
              isShowInput=true
              buttonText="showInput"
            />,
          );

        component |> ReactTestTool.createSnapshotAndMatch;
      });

      describe("submit", () => {
        test("click submit-button,the onSubmit func should be called", () => {
          open FileInput;

          let inputValue = "you can click submit after input value";
          let onSubmitFunc = SinonTool.createOneLengthStub(sandbox^);
          let state = {inputValue, isShowInput: true};

          FileInput.reducer(Some(onSubmitFunc), FileInput.Submit, state)
          |> ignore;

          onSubmitFunc |> expect |> toCalledWith([|inputValue|]);
        });
        test("submit trimed inputValue", () => {
          open FileInput;

          let onSubmitFunc = SinonTool.createOneLengthStub(sandbox^);
          let state = {inputValue: " aaa ", isShowInput: true};

          FileInput.reducer(onSubmitFunc, FileInput.Submit, state) |> ignore;

          onSubmitFunc |> expect |> toCalledWith([|"aaa"|]);
        });
      });
    });

    describe("deal with the specific case", () =>
      test(
        "if input value == '', click submit-button shouldn't handle onSubmit func",
        () => {
        open FileInput;

        let onSubmitFunc = SinonTool.createOneLengthStub(sandbox^);
        let state = {inputValue: "", isShowInput: true};

        FileInput.reducer(Some(onSubmitFunc), FileInput.Submit, state)
        |> ignore;

        onSubmitFunc |> expect |> not_ |> toCalled;
      })
    );
  });