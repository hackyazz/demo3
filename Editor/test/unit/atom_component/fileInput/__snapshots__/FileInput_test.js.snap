// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput test workflow click the showInput button, show the textarea and submit-button 1`] = `
<article
  className="wonder-file-input"
>
  <button
    onClick={[Function]}
  >
    showInput
  </button>
  <div>
    <textarea
      className="input-component file-input"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      submit
    </button>
  </div>
</article>
`;

exports[`FileInput test workflow key in text 1`] = `
<article
  className="wonder-file-input"
>
  <button
    onClick={[Function]}
  >
    showInput
  </button>
  <div>
    <textarea
      className="input-component file-input"
      onChange={[Function]}
      type="text"
      value="you can input value in textarea"
    />
    <button
      onClick={[Function]}
    >
      submit
    </button>
  </div>
</article>
`;

exports[`FileInput test workflow set showInput button text 1`] = `
<article
  className="wonder-file-input"
>
  <button
    onClick={[Function]}
  >
    showInput
  </button>
</article>
`;
